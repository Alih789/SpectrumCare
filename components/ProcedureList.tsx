import React from 'react';
import {useNavigation} from '@react-navigation/native';
import {
  FlatList,
  View,
  TouchableHighlight,
  Text,
  StyleSheet,
} from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {PrepInfoProps} from '../assets/customTypes';

type ProcedureListProps = {data: PrepInfoProps[]};

function ProcedureList({data}: ProcedureListProps) {
  return (
    <FlatList
      data={data.sort(compareObjectTitles)}
      renderItem={({item}) => <Route title={item.title} routeID={item.id} />}
      keyExtractor={item => item.id}
      style={styles.list}
    />
  );
}

type RouteProps = {title: string; routeID: string};

function Route({title, routeID}: RouteProps) {
  const navigation = useNavigation();

  return (
    <View>
      <TouchableHighlight
        activeOpacity={0.6}
        underlayColor="#DDDDDD"
        onPress={() => {
          navigation.navigate('Prep', {
            screen: 'Route',
            params: {
              routeID: routeID,
              routeTitle: title,
            },
          });
          // console.log("ROUTE ID IS: " + routeID)
        }}
        style={styles.touchable}>
        <View style={styles.route}>
          <Text style={styles.routetitle}>{title}</Text>
          <View style={styles.routebutton}>
            <Ionicons
              name={'arrow-forward-circle-outline'}
              size={40}
              color="white"
            />
          </View>
        </View>
      </TouchableHighlight>
    </View>
  );
}

const styles = StyleSheet.create({
  background: {
    backgroundColor: '#003a5d',
  },
  list: {
    height: '100%',
  },
  touchable: {
    backgroundColor: '#00b2e3',
    marginVertical: 5,
    marginHorizontal: 12,
    borderRadius: 10,
  },
  route: {
    backgroundColor: '#003a5d',
    paddingLeft: 20,
    flexDirection: 'row',
    flex: 2,
    justifyContent: 'space-between',
    alignContent: 'center',
    borderRadius: 10,
    borderTopColor: 'white',
    borderTopWidth: 1,
    borderBottomColor: 'white',
    borderBottomWidth: 1,
  },
  routetitle: {
    fontSize: 18,
    color: 'white',
    fontWeight: 'bold',
    alignSelf: 'center',
    flex: 1,
    flexWrap: 'wrap',
  },
  routebutton: {
    alignSelf: 'flex-end',
    padding: 10,
  },
});
export default ProcedureList;

function compareObjectTitles(a: PrepInfoProps, b: PrepInfoProps) {
  const nameA = a.title.toUpperCase(); // ignore upper and lowercase
  const nameB = b.title.toUpperCase(); // ignore upper and lowercase
  if (nameA < nameB) {
    return -1;
  }
  if (nameA > nameB) {
    return 1;
  }

  // names must be equal
  return 0;
}
