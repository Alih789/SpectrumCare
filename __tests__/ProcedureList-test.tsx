import 'react-native';
import React from 'react';
import ProcedureList from '../components/ProcedureList';
import renderer from 'react-test-renderer';

jest.mock('@react-navigation/native', () => ({
    useNavigation: jest.fn(),
}));

describe("ProcedureList", () => {
    const testProcedureData = require('../assets/testData/testProcedureData.json');

    test("renders ProcedureList", () => {
        const tree = renderer.create(
            <ProcedureList
                data={testProcedureData}
                data-testid="container-test-id"
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});