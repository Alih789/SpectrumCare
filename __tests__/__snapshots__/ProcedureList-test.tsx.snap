// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcedureList renders ProcedureList 1`] = `
<RCTScrollView
  data={
    [
      {
        "data": [
          {
            "category": "Radiology",
            "id": "mri",
            "pages": [
              {
                "bodyText": "There are many things to do before you go take an MRI.",
                "header": "Before Procedure:",
                "media": {
                  "content": "https://images.pexels.com/photos/6234634/pexels-photo-6234634.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
                  "isVideo": false,
                },
              },
              {
                "bodyText": "This is the sound that an MRI will make during the scan. WARNING: It is very loud!",
                "header": "During Procedure:",
                "media": {
                  "content": "4cSV5FTD1RI",
                  "isVideo": true,
                },
              },
              {
                "bodyText": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Aut doloremque totam quibusdam itaque voluptates dolorum laboriosam corrupti suscipit possimus aliquam, consequuntur odio ut quo nulla non sequi, sit, temporibus tempore. Lorem ipsum dolor sit, amet consectetur adipisicing elit. Id ipsa fuga aliquid, soluta delectus placeat, omnis voluptate consequatur in nulla itaque, odit eius ullam expedita? Quam eaque delectus asperiores rem! ",
                "header": "After Procedure:",
                "media": {
                  "content": "https://images.pexels.com/photos/6129644/pexels-photo-6129644.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
                  "isVideo": false,
                },
              },
            ],
            "title": "MRI",
          },
        ],
        "title": "Radiology",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "borderBottomColor": "white",
            "borderBottomWidth": 0.5,
            "marginHorizontal": 12,
          }
        }
      >
        <Text
          style={
            {
              "alignSelf": "center",
              "color": "white",
              "flex": 1,
              "flexWrap": "wrap",
              "fontFamily": "Figtree-Bold",
              "fontSize": 28,
              "marginBottom": 15,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 30,
            }
          }
        >
          Radiology
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View>
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#00b2e3",
              "borderRadius": 10,
              "marginHorizontal": 12,
            }
          }
        >
          <View
            style={
              {
                "alignContent": "center",
                "backgroundColor": "#003a5d",
                "borderBottomColor": "white",
                "borderBottomWidth": 0.5,
                "flex": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 20,
              }
            }
          >
            <Text
              style={
                {
                  "alignSelf": "center",
                  "color": "white",
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontFamily": "Figtree-Medium",
                  "fontSize": 22,
                }
              }
            >
              MRI
            </Text>
            <View
              style={
                {
                  "alignSelf": "flex-end",
                  "padding": 10,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "white",
                      "fontSize": 40,
                    },
                    undefined,
                    {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                ï„²
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
